Assets/Package/Runtime/Console/ConsoleGui.cs

```diff
@@ -144,14 +161,34 @@ private void EnsureInitialized()
             {
                 { inputSettings.ConfirmationKey, InvokeInputString },
                 { inputSettings.NextCommandKey, FetchNextCommand },
-                { inputSettings.PrevCommandKey, FetchPrevCommand }
+                { inputSettings.PrevCommandKey, FetchPrevCommand },
+                { inputSettings.AutocompleteKey, PerformAutocomplete }
             };
 
+            autocompleteHandler = new AutocompleteHandler();
             windowRect = styleSettings.InitialRect;
             scrollPosition = Vector2.zero;
             isInitialized = true;
         }
 
+        private void InputChanged()
+        {
+            UpdateAutocomplete();
+        }
+
+        private void UpdateAutocomplete()
+        {
+            if (string.IsNullOrEmpty(currentInput))
+            {
+                currentAutocompleteMatch = string.Empty;
+                return;
+            }
+
+            var splitInput = currentInput.Split(' ');
+            autocompleteHandler.RefreshOptions(consoleManager, splitInput, splitInput.Length - 1);
+            currentAutocompleteMatch = autocompleteHandler.GetBestMatch(splitInput.Last());
```

Comment:
Myślę, że te dwie metody możnaby zmergować w jedną metodę.

Reply:
I generalnie split input powinien być imho robiony w metodzie AutocompleteHandlera w tym momencie. Jak to wszystko się przerzuci tam to zostanie one-liner, którego można takim razie przerzucić do metody OnInputChange